ecs-cli/modules/config/readwriter_v1.go

```diff
@@ -58,56 +59,108 @@ func NewReadWriter() (*YAMLReadWriter, error) {
 	return &YAMLReadWriter{destination: dest}, nil
 }
 
-func readINI(dest *Destination) (*CLIConfig, map[interface{}]interface{}, error) {
+func readINI(dest *Destination) (*CLIConfig, error) {
 	iniReadWriter, err := NewINIReadWriter(dest)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 	return iniReadWriter.GetConfig()
 }
 
-func readYAML(yamlPath string, configMap map[interface{}]interface{}, cliConfig *CLIConfig) error {
-	// convert yaml to CliConfig
-	dat, err := ioutil.ReadFile(yamlPath)
+// readClusterConfig does all the work to read and parse the yaml cluster config
+func readClusterConfig(path string, clusterConfigKey string, cliConfig *CLIConfig) error {
+	// read cluster file
+	dat, err := ioutil.ReadFile(path)
+	if err != nil {
+		return errors.Wrap(err, "Failed to read config file: "+path)
+	}
+
+	config := ClusterConfig{Clusters: make(map[string]Cluster)}
+	if err = yaml.Unmarshal(dat, &config); err != nil {
+		return errors.Wrap(err, "Failed to parse yaml file: "+path)
+	}
+
+	// get the correct cluster
```

Comment:
Ughh. Because I had it as that, but if you go back a few diffs one of you guys commented on the PR and said that it was bad to mutate an argument. So I changed it to this...